lib/matplotlib/axes/_subplots.py

```diff
@@ -172,7 +172,7 @@ def _make_twin_axes(self, *kl, **kwargs):
         Make a twinx axes of self. This is used for twinx and twiny.
         """
         from matplotlib.projections import process_projection_requirements
-        if 'sharex' in kwargs and 'sharey' in kwargs:
+        if kwargs["sharex"] is not self and kwargs["sharey"] is not self:
```

Comment:
I think this needs to be a separate check as well as the one it has replaced right? The currently one checks that both x and y aren't shared at the same time, and you want to check that we're not trying to share an axis with itself.

Reply:
It will also give a `KeyError` if "sharex" and "sharey" aren't in `kwargs`.

Reply:
argh... ok, tuning